code/modules/ruins/objects_and_mobs/ash_walker_den.dm

```diff
@@ -55,6 +73,17 @@
 				else
 					SEND_SIGNAL(L, COMSIG_ADD_MOOD_EVENT, "oogabooga", /datum/mood_event/sacrifice_bad)
 
+/obj/structure/lavaland/ash_walker/proc/remake_walker(var/datum/mind/oldmind, var/oldname)
+	var/mob/living/carbon/human/M = new /mob/living/carbon/human(get_step(loc, pick(GLOB.alldirs)))
+	M.set_species(/datum/species/lizard/ashwalker)
+	M.real_name = oldname
+	M.underwear = "Nude"
+	M.update_body()
+	oldmind.transfer_to(M)
+	M.mind.grab_ghost()
+	to_chat(M, "<b>You have been pulled back from beyond the grave, with a new body and renewed purpose. Glory to the Necropolis!</b>")
```

Comment:
might wanna specifically say that the respawned ash walker keeps their memories

Reply:
I was hoping it would be implied well enough, since the revived Walker gets the same name as their old body as well as the same IC notes, but I could add to the text here a bit.